### PR TITLE
Document the type and source of all metadata used and requested from optimizer

### PR DESCRIPTION
**Pull request by armenatzoglou**
@hlinnaka @gcaragea @entong please take a look.

### CODE REVIEW

**Comment by entong** [comment]
LGTM

**Comment by hlinnaka** [comment]
LGTM

File: src/backend/gpopt/README [hunk 1]
```diff
@@ -0,0 +1,25 @@
+===================================================================================================
+This document describes the type and source of all metadata used and requested from optimizer. GPDB
+collects all these data using GPDB function wrappers. 
+Copyright (c) 2015, Pivotal Software, Inc.
+---------------------------------------------------------------------------------------------------
+
+
+Metadata Type (neucrates/md/IMDId.h) 	|	Includes		|	Source
```
Comment (gcaragea) [inline src/backend/gpopt/README]: neucrates -> naucrates

**Comment by gcaragea** [comment]
LGTM

**Comment by armenatzoglou** [comment]
Pushed code. Closing.

**Pull request closed without merging**